Fix Incorrect Runs Being Processed After Loading Batch

cailafinn opened the pull request:
**Description of work.**
When loading a new batch the selection could update on the table but the model wouldn't track the changed selection. This sends a notification all the way down to the table to update the current selection so the correct runs/groups are processed. 

**To test:**
1. Open the Refl GUI
2. Load a batch from the sample data
3. Select group 2
4. Load the same batch file again
5. Click process
6. Only the selected row should be processed


Fixes #28001 


---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

Commit:
Update selection after loading

When loading a new batch the selection could update on the table but the
model wouldn't track the changed selection. This sends a notification
all the way down to the table to update the current selection.

RE #28001

Commit:
Add unit tests for new functionality

RE #28001

Commit:
Add to release notes

RE #28001

sf1919 commented on docs/source/release/v6.3.0/reflectometry.rst:
```diff
@@ -15,5 +15,6 @@ Bugfixes
 --------
 - The Add Row button is now disabled in the Experiment settings tab when runs are being processed.
 - Invalid per-angle defaults values that are saved into a batch will now be marked as invalid when loaded back into the GUI.
+- Fixed un-selected runs/groups being processed after loading a batch.
```
Should this be groups rather than runs? When I was testing the PR I found it only worked if I selected the second group. If I selected an individual run it would run on all runs.

Commit:
Improve release note

Does not actually affect the selection of rows after loading a batch as
these do not ever seem to stay selected after loading. Only groups are
effected.

RE #28001

sf1919 reviewed (approved):
I have tested the PR and it does as expected. I have also reviewed the code and it looks like a neat way of resolving the issue. Also pleased to see a suitable test. Thank you for updating the release note as requested.

I therefore approve this PR.

DavidFair reviewed (approved):
LGTM :tada:
